### ISSUE TITLE
How to freeze a layer?

### ISSUE THREAD

**Issue by arashno** [torch/nn#394]
Hi everyone.
I am wondering how can I freeze one or more layers of a network during training?
Thanks

**Comment by soumith** [comment]
You can set that layer's accGradParameters function to an empty function.

**Comment by SunnyWay** [comment]
Hi @soumith. 
Is it possible to implement a Freeze container which freezes the modules inside? I think it will be more convenient than setting the layer's accGradParameters function to an empty function.

**Comment by dno89** [comment]
Hi everyone,
does the proposed solution work even if you flatten the parameters' tensor for training with optim?

**Comment by joeyhng** [comment]
It won't work if you have weight decay. I would make parameters() function to an empty function.

**Comment by SunnyWay** [comment]
@dno89 I implemented a Container which shared the idea of @joeyhng. You can find it on torch forum: https://groups.google.com/forum/#!topic/torch7/NMb4DcUaKdg

**Comment by dno89** [comment]
@SunnyWay nice solution, thank you!
